### PR TITLE
[Validator] Add `Cascade` constraint documentation

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fixes https://github.com/symfony/symfony-docs/issues/14015

### CODE REVIEW

File: reference/constraints/Cascade.rst [hunk 1]
```diff
@@ -0,0 +1,120 @@
+Cascade
+=======
+
+.. versionadded:: 5.2
+
+    The :class:`Symfony\\Component\\Validator\\Constraints\\Cascade` was
+    introduced in Symfony 5.2 and requires PHP 7.4.
```
Comment (MrYamous) [inline reference/constraints/Cascade.rst:7]: What do you think about using a caution block for PHP version restriction ? I think this information is more important

Comment (alexandre-daubois) [reply reference/constraints/Cascade.rst:7]: Actually, versionadded blocks are removed by maintainers when upmerging the branch. Caution blocks are not, and as this is only a relevant info for 5.4, I would say it should stay in this block 🙂

**Comment by javiereguiluz** [comment]
I wasn't aware of this missing constraint. Alex, we can't thank you enough for your invaluable work contributing all these missing features 🙇

**Comment by alexandre-daubois** [comment]
@javiereguiluz my pleasure :bow: I was shocked I never heard of it either!

**Pull request merged**